Extract opening a pre_warm connection

JosiahWI opened the pull request:
This extracts opening a pre_warm connection into a few functions and moves PreWarmManager from proxy/http into iocore/net.

JosiahWI committed:
Extract opening a pre_warm connection

This extracts opening a pre_warm connection into a few functions
and moves PreWarmManager from proxy/http into iocore/net. This
makes the logic easier to follow, and the PreWarmManager is used
primarily in iocore; moving it there brings us closer to getting
rid of the cyclic dependency between iocore/net and proxy/http.

bneradt commented on proxy/http/HttpSM.cc:
```diff
@@ -5258,6 +5257,56 @@ HttpSM::get_outbound_sni() const
+  handleEvent(NET_EVENT_OPEN, netvc);
+}
+
+>>>>>>> 1a1ddc209 (Extract opening a pre_warm connection)
```
Looks like a left over merge conflict line.

bneradt reviewed (changes requested):
Looks fine. Be sure to explain the reasoning for the refactor/relocation clearly in your apache/trafficserver PR.

moonchen committed:
Fix a compiler warning in P_SNIActionPerformer.h (#10071)

Put a member variable behind #ifdef to prevent unused variable warning.

masaori335 committed:
LSan: Fix memory leak of test_HostFile (#10064)

JosiahWI committed:
Clean up link libraries for traffic_server target (#10074)

shukitchan committed:
Use of uninitialized variables reported by oss-fuzz (#10068)

* initialize variables reported by oss-fuzz

* fix according to comments

JosiahWI committed:
Fix missing PCRE includes in CMake by linking to tscore (#10073)

* Add transitive PCRE deps to lots of targets

This fixes all the PCRE include errors I have noticed on Mac.

* Link everything that needs tscore to tscore

This is a step towards getting the dependency tree fixed. And it
fixes the missing pcre includes.

* Fix build on macOS

- Use -undefined dynamic_lookup to ignore unresolved dependencies which are located inside traffic_server
- Add some missing dependencies
- Break a cyclic dependency between tscppapi and tscore

* Use target_link_options instead of LINK_FLAGS

* Fix indentation

---------

Co-authored-by: Mo Chen <mochen@apache.org>

Commit:
fix: require RSRC_CLIENT_REQUEST_HEADERS in ConditionMethod (#9976)
